src/parser.c

```diff
@@ -364,20 +364,33 @@ char* parse_insert_values (pTHX_ HV* state, register char* p, AV* ret) {
       p++;
 
       I32 column_id = 0;
+#if DEBUG_MODE
+      int is_binary_literal = 0;
+#endif
       while (*p != '\0' && *p != ')') {
         // get column name
         SV** column_ref = XSUTIL_AV_FETCH(columns, column_id);
         if (! column_ref) croak("Cannot fetch columns[%d]", column_id);
         SV* column = *column_ref;
         DEBUG_OUT("key: %s\n", SvPV_nolen(column));
 
+      Again:
         // extract and store value
         if (IS_NULL_STR(p)) {
           // null value
           p += 4;
           DEBUG_OUT("value: (NULL)\n");
           XSUTIL_HV_STORE_ENT_NOINC(row, column, &PL_sv_undef);
         }
+        else if (IS__binary(p)) {
+          // skip '_binary' character set introducer for string literals
+          p += 7;
+          SKIP_WSPACE(p);
+#if DEBUG_MODE
+          is_binary_literal = 1;
+#endif
+          goto Again;
```

Comment:
It can accept an invalid string. eg: `_binary_binary"foo"`. so I think it is not a good idea.